onnxruntime/contrib_ops/webgpu/quantization/matmul_nbits.h

```diff
@@ -12,41 +12,44 @@ namespace webgpu {
 
 using namespace onnxruntime::webgpu;
 
-class MatMulNBitsProgram final : public Program<MatMulNBitsProgram> {
+class MatMulNBitsWideTileProgram final : public Program<MatMulNBitsWideTileProgram> {
  public:
-  MatMulNBitsProgram(uint32_t output_number, uint32_t block_size, uint32_t tile_m, int components_b, bool has_zero_points, bool use_subgroup) : Program{"MatMulNBits"},
-                                                                                                                                                output_number_{output_number},
-                                                                                                                                                block_size_{block_size},
-                                                                                                                                                tile_m_{tile_m},
-                                                                                                                                                components_b_{components_b},
-                                                                                                                                                has_zero_points_{has_zero_points},
-                                                                                                                                                use_subgroup_(use_subgroup) {
-  }
+  MatMulNBitsWideTileProgram(bool has_zero_points, uint32_t tile_m, uint32_t tile_n, uint32_t nbits)
+      : Program{"MatMulNBitsWideTileProgram"}, has_zero_points_{has_zero_points}, tile_m_(tile_m), tile_n_(tile_n), nbits_(nbits) {}
 
   Status GenerateShaderCode(ShaderHelper& sh) const override;
-  WEBGPU_PROGRAM_DEFINE_UNIFORM_VARIABLES({"block_size", ProgramUniformVariableDataType::Uint32});
+  WEBGPU_PROGRAM_DEFINE_UNIFORM_VARIABLES({"block_size", ProgramUniformVariableDataType::Uint32},
+                                          {"zero_blocks_per_col", ProgramUniformVariableDataType::Uint32},
+                                          {"num_N_tile", ProgramUniformVariableDataType::Uint32},
+                                          {"num_M_tile", ProgramUniformVariableDataType::Uint32});
 
  private:
-  uint32_t output_number_;
-  uint32_t block_size_;
-  uint32_t tile_m_;
-  int components_b_;
   bool has_zero_points_;
-  bool use_subgroup_;
+  uint32_t tile_m_;
+  uint32_t tile_n_;
+  uint32_t nbits_;
 };
 
-class MatMulNBitsWideTileProgram final : public Program<MatMulNBitsWideTileProgram> {
+class MatMulNBitsBlockWiseProgram final : public Program<MatMulNBitsBlockWiseProgram> {
```

Comment:
nit: Not sure what BLockWise means. Can these simply be MatMulNBitsSmallMProgram